adapter-pinecone/src/main/java/io/nosqlbench/adapter/pinecone/opdispensers/PineconeUpdateOpDispenser.java

```diff
@@ -0,0 +1,173 @@
+                metadata_map.put(key, targetval);
+            };
+            Map<String, Object> metadata_values_map = mapLongFunction.apply(l);
+            metadata_values_map.forEach(stringToValue);
+            return UpdateRequest.newBuilder().getSetMetadataBuilder().putAllFields(metadata_map).build();
+        }).orElse(null);
+    }
+
+    /**
+     * @param op The ParsedOp used to build the Request
+     * @return A function that will take a long (the current cycle) and return a Pinecone UpdateRequest Builder
+     *
+     * The pattern used here is to accommodate the way Request types are constructed for Pinecone.
+     * Requests use a Builder pattern, so at time of instantiation the methods should be chained together.
+     * For each method in the chain a function is created here and added to the chain of functions
+     * called at time of instantiation.
+     *
+     * The Metadata and SparseValues objects used by the UpdateRequest are sufficiently sophisticated in their own
+     * building process that they have been broken out into separate methods. At runtime they are built separately
+     * and then added to the build chain by the builder returned by this method.
+     */
+    private LongFunction<UpdateRequest.Builder> createUpdateRequestFunction(ParsedOp op) {
+        LongFunction<UpdateRequest.Builder> rFunc = l -> UpdateRequest.newBuilder();
+
+        Optional<LongFunction<String>> nFunc = op.getAsOptionalFunction("namespace", String.class);
+        if (nFunc.isPresent()) {
+            LongFunction<UpdateRequest.Builder> finalFunc = rFunc;
+            LongFunction<String> af = nFunc.get();
+            rFunc = l -> finalFunc.apply(l).setNamespace(af.apply(l));
+        }
+
+        Optional<LongFunction<String>> iFunc = op.getAsOptionalFunction("id", String.class);
+        if (iFunc.isPresent()) {
+            LongFunction<UpdateRequest.Builder> finalFunc = rFunc;
+            LongFunction<String> af = iFunc.get();
+            rFunc = l -> finalFunc.apply(l).setId(af.apply(l));
+        }
+
+        Optional<LongFunction<String>> vFunc = op.getAsOptionalFunction("values", String.class);
+        if (vFunc.isPresent()) {
+            LongFunction<UpdateRequest.Builder> finalFunc = rFunc;
+            LongFunction<String> af = vFunc.get();
+            LongFunction<ArrayList<Float>> alf = l -> {
+                String[] vals = af.apply(l).split(",");
+                ArrayList<Float> fVals = new ArrayList<>();
+                for (String val : vals) {
+                    fVals.add(Float.valueOf(val));
+                }
+                return fVals;
+            };
+            rFunc = l -> finalFunc.apply(l).addAllValues(alf.apply(l));
+        }
+
+        return rFunc;
+    }
+
+    @Override
+    public PineconeOp apply(long value) {
+        UpdateRequest.Builder urb = updateRequestFunc.apply(value);
```

Comment:
I'm sure there is a way to collapse these conditionals into functional form, but it's not important enough to block the PR.